src/vhost_user/slave_req_handler.rs

```diff
@@ -537,7 +537,6 @@ impl<S: VhostUserSlaveReqHandler> SlaveReqHandler<S> {
         let vflag = VhostUserVirtioFeatures::PROTOCOL_FEATURES.bits();
         let pflag = VhostUserProtocolFeatures::REPLY_ACK;
         if (self.virtio_features & vflag) != 0
-            && (self.acked_virtio_features & vflag) != 0
```

Comment:
It would be great to document the right behavior and why we do not follow the right behavior, to follow a qemu "feature":)

Reply:
Well as far as I'm aware the canonical reference for vhost-user is kept in the QEMU source tree and part of it's [interoperability documentation](https://qemu.readthedocs.io/en/latest/interop/vhost-user.html#protocol-features). I think the key is the wording for the GET/SET protocol features:

    ``VHOST_USER_GET_PROTOCOL_FEATURES``
      :id: 15
      :equivalent ioctl: ``VHOST_GET_FEATURES``
      :master payload: N/A
      :slave payload: ``u64``

      Get the protocol feature bitmask from the underlying vhost
      implementation.  Only legal if feature bit
      ``VHOST_USER_F_PROTOCOL_FEATURES`` is present in
      ``VHOST_USER_GET_FEATURES``.

    .. Note::
       Slave that reported ``VHOST_USER_F_PROTOCOL_FEATURES`` must
       support this message even before ``VHOST_USER_SET_FEATURES`` was
       called.

    ``VHOST_USER_SET_PROTOCOL_FEATURES``
      :id: 16
      :equivalent ioctl: ``VHOST_SET_FEATURES``
      :master payload: ``u64``

      Enable protocol features in the underlying vhost implementation.

      Only legal if feature bit ``VHOST_USER_F_PROTOCOL_FEATURES`` is present in
      ``VHOST_USER_GET_FEATURES``.

    .. Note::
       Slave that reported ``VHOST_USER_F_PROTOCOL_FEATURES`` must support
       this message even before ``VHOST_USER_SET_FEATURES`` was called.

To me the note kind of implies that the negotiation of the protocol is independent of the final SET_FEATURES state. I think if anything we would want to add more clarity to the reference document.

Reply:
Thanks, seems I missed the "Note" section when implementing the vhost crate:(

Reply:
could you please help to update the PR? We have just merged a big PR, so rebase is needed:)

Reply:
Will do, just send a clarifcation patch to the vhost-user spec to be super explicit first.